src/connector/src/sink/encoder/proto.rs

```diff
@@ -484,26 +529,360 @@ mod tests {
             ]))),
             Some(ScalarImpl::List(ListValue::from_iter([4, 0, 4]))),
             Some(ScalarImpl::Timestamptz(Timestamptz::from_micros(3))),
+            Some(ScalarImpl::Map(
+                MapValue::try_from_kv(
+                    ListValue::from_iter(["a", "b"]),
+                    ListValue::from_iter([1, 2]),
+                )
+                .unwrap(),
+            )),
+            {
+                let mut struct_array_builder = StructArrayBuilder::with_type(
+                    2,
+                    DataType::Struct(StructType::new(vec![
+                        ("id", DataType::Int32),
+                        ("name", DataType::Varchar),
+                    ])),
+                );
+                struct_array_builder.append(Some(
+                    StructValue::new(vec![
+                        Some(ScalarImpl::Int32(1)),
+                        Some(ScalarImpl::Utf8("x".into())),
+                    ])
+                    .as_scalar_ref(),
+                ));
+                struct_array_builder.append(Some(
+                    StructValue::new(vec![
+                        Some(ScalarImpl::Int32(2)),
+                        Some(ScalarImpl::Utf8("y".into())),
+                    ])
+                    .as_scalar_ref(),
+                ));
+                Some(ScalarImpl::Map(
+                    MapValue::try_from_kv(
+                        ListValue::from_iter(["a", "b"]),
+                        ListValue::new(struct_array_builder.finish().into()),
+                    )
+                    .unwrap(),
+                ))
+            },
         ]);
 
         let encoder =
             ProtoEncoder::new(schema, None, descriptor.clone(), ProtoHeader::None).unwrap();
         let m = encoder.encode(row).unwrap();
-        let encoded: Vec<u8> = m.ser_to().unwrap();
-        assert_eq!(
-            encoded,
-            // Hint: write the binary output to a file `test.binpb`, and view it with `protoc`:
-            // ```
-            // protoc --decode_raw < test.binpb
-            // protoc --decode=all_types.AllTypes all-types.proto < test.binpb
-            // ```
-            [
-                9, 0, 0, 0, 0, 0, 0, 17, 64, 21, 0, 0, 96, 64, 24, 22, 32, 23, 56, 48, 93, 26, 0,
-                0, 0, 97, 27, 0, 0, 0, 0, 0, 0, 0, 104, 1, 114, 10, 82, 105, 115, 105, 110, 103,
-                87, 97, 118, 101, 122, 2, 190, 239, 138, 1, 2, 8, 1, 146, 1, 3, 4, 0, 4, 186, 1, 3,
-                16, 184, 23
-            ]
-        );
```

Comment:
serialize map seems to be nondeterministic, so we can't test it in this way any more